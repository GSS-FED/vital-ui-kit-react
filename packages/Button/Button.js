// @flow

import * as React from 'react';

import ButtonElement from './styled';
import { defaultTheme } from './constant';

type Nature = 'default' | 'primary' | 'success' | 'info' | 'alarm' | 'warning';

type Size = 'xlarge' | 'large' | 'medium' | 'small' | 'xsmall';

type Props = {
  /** The content of the button */
  children?: React.Node,
  /** on click event */
  onClick?: () => void,
  /** 6 nature state */
  nature?: Nature,
  /** Flat style */
  flat?: boolean,
  /** Subtle style */
  subtle?: boolean,
  /** Light style */
  light?: boolean,
  /** Link style */
  link?: boolean,
  /** Button size */
  size?: Size,
  theme?: any,
  /** Circle button */
  circle?: boolean,
  /** Selected state */
  selected?: boolean,
}

const Button = ({ children, onClick, ...props }: Props) => (
  <ButtonElement onClick={onClick} {...props}>
    {children}
  </ButtonElement>
);


Button.defaultProps = {
  children: 'Button',
  nature: 'default',
  theme: defaultTheme,
  size: 'medium',
  light: false,
  flat: false,
  link: false,
  subtle: false,
  selected: false,
  onClick: () => {},
};

export default Button;
