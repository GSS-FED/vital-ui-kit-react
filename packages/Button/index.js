export { default as Button } from './Button';
export { default as ButtonGroup } from './ButtonGroup';

/**
 * @render react
 * @name ButtonGroup
 * @description Vital Buttons Group
 * @example
 * <ButtonGroup>
 *   <Button nature="default" size="xlarge">Default</Button>
 *   <Button nature="primary" size="large">Default</Button>
 *   <Button nature="success" size="medium">Default</Button>
 *   <Button nature="info" size="small">Default</Button>
 *   <Button nature="alarm" size="xsmall">Default</Button>
 *   <Button nature="warning">Default</Button>
 *   <Button flat>Default</Button>
 *   <Button subtle>Default</Button>
 *   <Button light>Default</Button>
 *   <Button link>Default</Button>
 * </ButtonGroup>
 */
