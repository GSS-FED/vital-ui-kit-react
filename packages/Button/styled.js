import styled from 'styled-components';
import { darken, lighten } from 'polished';

import { Styles } from '../Base/';
import { buttonSizes } from './constant';

const { colors } = Styles;

const ButtonElement = styled.button`
  position: relative;
  display: inline-block;
  background: ${props => props.theme.Button[props.nature].background};
  color: ${props => props.theme.Button[props.nature].color};
  cursor: pointer;
  border-width: 1px;
  border-style: solid;
  border-color: ${colors.primaryList[3]};
  font-size: ${({ size }) => buttonSizes[size]['font-size']};
  border-radius: ${({ size }) => buttonSizes[size]['border-radius']};
  padding: ${({ size }) => buttonSizes[size].padding};
  line-height: 1;
  font-weight: normal;
  margin: 0;
  outline: none;
  padding: ${({ size }) => buttonSizes[size].padding};
  box-sizing: border-box;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;

  &:hover {
    background: ${colors.primaryList[2]};
  }

  &:active {
    background: ${colors.primaryList[4]};
  }

  ${({ subtle, selected }) => subtle &&`
    background: ${selected ? colors.primaryList4 : 'transparent'};
    color: ${colors.primaryList7};
    border: 1px solid transparent;

    &:hover {
      background: ${colors.primaryList[2]};
    }
  `};

  ${props => props.flat &&`
   background: ${props.nature === 'default'
       ? colors.primaryList[1]
       : props.theme.Button[props.nature].color};
   color: ${props.nature === 'default'
       ? props.theme.Button.default.color
       : colors.white};
   border-color: ${props.nature === 'default'
       ? colors.primaryList[1]
       : props.theme.Button[props.nature].color};

   &:hover {
     background: ${props.nature === 'default'
         ? colors.primaryList[2]
         : lighten(0.1, props.theme.Button[props.nature].color)};
     border-color: ${props.nature === 'default'
         ? colors.primaryList[2]
         : lighten(0.1, props.theme.Button[props.nature].color)};
   }
   &:active {
     background: ${props.nature === 'default'
         ? colors.primaryList[4]
         : darken(0.12, props.theme.Button[props.nature].color)};
     border-color: ${props.nature === 'default'
         ? colors.primaryList[4]
         : darken(0.12, props.theme.Button[props.nature].color)};
   }
  `};

  ${props => props.light &&`
   background: ${colors.white};
   color: ${props.theme.Button[props.nature].color};

   &:hover {
     background: ${colors.primaryList[2]};
   }
  `}

  ${({ link }) => link &&`
   background: transparent;
   color: ${colors.primary};
   border: 1px solid transparent;

   &:hover {
     background: transparent;
     color: ${lighten(0.1, colors.primary)};
   }`}

   ${({ link, dark }) => (link && dark) &&`
      color: ${colors.primaryList[6]};

    &:hover {
      color: ${lighten(0.1, colors.primaryList[6])};
    }
   `}

   ${({ underline }) => underline &&`
      &:hover {
        text-decoration: underline;
      }
   `}

   ${({ circle }) => circle &&`
      padding: 0;
      border-radius: 100%;
      width: 2.066rem;
      line-height: 2.066rem;
      height: 2.066rem;
   `}

  &[disabled] {
    color: #93a8ce;
    border-color: #eaeff8;
    background-color: #f7f9fe;
    cursor: not-allowed;
    pointer-events: none;
  }
`;

export default ButtonElement;
