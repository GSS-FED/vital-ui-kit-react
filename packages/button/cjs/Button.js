'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _ButtonGroup = require('./ButtonGroup');

var _ButtonGroup2 = _interopRequireDefault(_ButtonGroup);

var _styled = require('./styled');

var _styled2 = _interopRequireDefault(_styled);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * 
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Copyright Â© 2018 Galaxy Software Services https://github.com/GSS-FED/vital-ui-kit-react
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * MIT license
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */

/**
 * @render react
 * @name Button
 * @description Button component with group
 * @example
 * <Button.Group>
 *   <Button nature="default" size="xlarge">Default</Button>
 *   <Button nature="primary" size="large">Default</Button>
 *   <Button nature="success" size="medium">Default</Button>
 *   <Button nature="info" size="small">Default</Button>
 *   <Button nature="alarm" size="xsmall">Default</Button>
 *   <Button nature="warning">Default</Button>
 *   <Button flat>Default</Button>
 *   <Button subtle>Default</Button>
 *   <Button light>Default</Button>
 *   <Button link>Default</Button>
 * </Button.Group>
 */

var Button = function (_React$Component) {
  _inherits(Button, _React$Component);

  function Button() {
    _classCallCheck(this, Button);

    return _possibleConstructorReturn(this, (Button.__proto__ || Object.getPrototypeOf(Button)).apply(this, arguments));
  }

  _createClass(Button, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          children = _props.children,
          className = _props.className,
          style = _props.style,
          onClick = _props.onClick,
          props = _objectWithoutProperties(_props, ['children', 'className', 'style', 'onClick']);

      return React.createElement(
        _styled2.default,
        _extends({
          className: (0, _classnames2.default)('Vital_Button', className),
          style: style,
          onClick: onClick
        }, props),
        children
      );
    }
  }]);

  return Button;
}(React.Component);

Button.defaultProps = {
  children: '',
  nature: 'default',
  size: 'medium',
  circle: false,
  light: false,
  flat: false,
  link: false,
  subtle: false,
  selected: false,
  onClick: function onClick() {},
  style: undefined,
  className: ''
};
Button.Group = _ButtonGroup2.default;
exports.default = Button;