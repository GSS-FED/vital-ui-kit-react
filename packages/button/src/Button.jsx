/**
 * @flow
 * Copyright Â© 2018 Galaxy Software Services https://github.com/GSS-FED/vital-ui-kit-react
 * MIT license
 */

import * as React from 'react';
import cn from 'classnames';

import ButtonGroup from './ButtonGroup';
import ButtonElement from './styled';

type Nature =
  | 'default'
  | 'primary'
  | 'success'
  | 'info'
  | 'alarm'
  | 'warning';

type Size = 'xlarge' | 'large' | 'medium' | 'small' | 'xsmall';

type Props = {
  /** `vital_button` */
  className?: string,
  /** The content of the button */
  children?: React.Node,
  /** on click event */
  onClick?: () => void,
  /** 6 nature state */
  nature?: Nature,
  /** Flat style */
  flat?: boolean,
  /** Subtle style */
  subtle?: boolean,
  /** Light style */
  light?: boolean,
  /** Link style */
  link?: boolean,
  /** Button size */
  size?: Size,
  /** Circle button */
  circle?: boolean,
  /** Selected state */
  selected?: boolean,
  style?: CSSStyleDeclaration,
};

/**
 * @render react
 * @name Button
 * @description Button component with group
 * @example
 * <Button.Group>
 *   <Button nature="default" size="xlarge">Default</Button>
 *   <Button nature="primary" size="large">Default</Button>
 *   <Button nature="success" size="medium">Default</Button>
 *   <Button nature="info" size="small">Default</Button>
 *   <Button nature="alarm" size="xsmall">Default</Button>
 *   <Button nature="warning">Default</Button>
 *   <Button flat>Default</Button>
 *   <Button subtle>Default</Button>
 *   <Button light>Default</Button>
 *   <Button link>Default</Button>
 * </Button.Group>
 */

class Button extends React.Component<Props> {
  static defaultProps = {
    children: '',
    nature: 'default',
    size: 'medium',
    circle: false,
    light: false,
    flat: false,
    link: false,
    subtle: false,
    selected: false,
    onClick: () => {},
    style: undefined,
    className: '',
  };

  static Group = ButtonGroup;

  render() {
    const {
      children,
      className,
      style,
      onClick,
      ...props
    } = this.props;
    return (
      <ButtonElement
        className={cn('vital__button', className)}
        style={style}
        onClick={onClick}
        {...props}
      >
        {children}
      </ButtonElement>
    );
  }
}

export default Button;
