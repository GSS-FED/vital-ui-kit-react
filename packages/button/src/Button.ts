/**
 * Copyright Â© 2018 Galaxy Software Services https://github.com/GSS-FED/vital-ui-kit-react
 * MIT license
 */

import { defaultTheme } from '@vital-ui/react-theme';
import { BoxProps } from '@vital-ui/react-utils';
import cn from 'classnames';
import styled from 'styled-components';
import { BuiltinSize, ButtonSize, Nature } from './constant';
import { natureColor } from './utils';
import { boxStyle } from '@vital-ui/react-utils';

function builtInOrTheme(
  builtin: BuiltinSize | undefined,
  theme: any,
) {
  return builtin ? builtin : theme.button;
}

export interface ButtonProps extends BoxProps {
  builtinSize?: BuiltinSize;
  /** 6 nature state */
  nature?: Nature;
  /** Button size */
  size?: ButtonSize;
  disabled?: boolean;
}

export const Button = styled('button').attrs(props => ({
  type: 'button',
  className: cn('vital__button', props.className),
  theme: defaultTheme,
  nature: 'default' as Nature,
  size: 'medium' as ButtonSize,
  selected: false,
  disabled: false,
}))<ButtonProps>`
  position: relative;
  background: ${props => props.theme.button.default.bg};
  color: ${({ theme, nature }) => natureColor(theme)[nature]};
  cursor: pointer;
  border-width: 1px;
  border-style: solid;
  border-color: ${props => props.theme.borderColor};
  font-size: ${({ builtinSize, theme, size }) =>
    builtInOrTheme(builtinSize, theme)[size].fontSize};
  border-radius: ${({ builtinSize, theme, size }) =>
    builtInOrTheme(builtinSize, theme)[size].borderRadius};
  padding: ${({ builtinSize, theme, size }) =>
    builtInOrTheme(builtinSize, theme)[size].padding};
  line-height: 1;
  font-weight: normal;
  margin: 0;
  outline: none;
  box-sizing: border-box;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  &:hover {
    background: ${props => props.theme.button.default.hoverBg};
  }

  &:active {
    background: ${props => props.theme.button.default.activeBg};
  }

  &[disabled] {
    color: ${({ theme }) => theme.button.disabled.color};
    border-color: ${({ theme }) => theme.button.disabled.borderColor};
    background: ${({ theme }) => theme.button.disabled.bg};
    cursor: not-allowed;
    pointer-events: none;
  }

  ${boxStyle};
  ${props => props.theme.button.buttonStyle};
`;
