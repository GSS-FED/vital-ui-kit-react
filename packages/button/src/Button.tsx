/**
 * Copyright Â© 2018 Galaxy Software Services https://github.com/GSS-FED/vital-ui-kit-react
 * MIT license
 */

import * as React from 'react';
import cn from 'classnames';
import { Nature, ButtonSize, BuiltinSize } from './constant';

import styled, { css, ThemedStyledProps } from 'styled-components';
import { darken, lighten } from 'polished';
import { defaultTheme } from '@vital-ui/react-theme';
import { superBoxStyle, BoxProps } from '@vital-ui/react-utils';

interface ButtonElementProps extends BoxProps {
  nature: Nature;
  size: ButtonSize;
  subtle?: boolean;
  selected?: boolean;
  flat?: boolean;
  light?: boolean;
  link?: boolean;
  dark?: boolean;
  underline?: boolean;
  circle?: boolean;
  builtinSize?: BuiltinSize;
}

type StyledButtonProps = ThemedStyledProps<ButtonElementProps, any>;

function builtInOrTheme(
  builtin: BuiltinSize | undefined,
  theme: any,
) {
  return builtin ? builtin : theme.button;
}

export const natureColor = (theme: any) => ({
  default: theme.colors.secondary700,
  primary: theme.colors.primary,
  success: theme.colors.success,
  info: theme.colors.info,
  alarm: theme.colors.alarm,
  warning: theme.colors.warning,
});

export const buttonBaseStyle = ({
  theme,
  size,
  nature,
  builtinSize,
}: StyledButtonProps) =>
  css`
    position: relative;
    background: ${theme.button.default.bg};
    color: ${natureColor(theme)[nature]};
    cursor: pointer;
    border-width: 1px;
    border-style: solid;
    border-color: ${theme.borderColor};
    font-size: ${builtInOrTheme(builtinSize, theme)[size].fontSize};
    border-radius: ${builtInOrTheme(builtinSize, theme)[size]
      .borderRadius};
    padding: ${builtInOrTheme(builtinSize, theme)[size].padding};
    line-height: 1;
    font-weight: normal;
    margin: 0;
    outline: none;
    box-sizing: border-box;
    vertical-align: middle;
    text-align: center;
    text-decoration: none;
    &:hover {
      background: ${theme.button.default.hoverBg};
    }

    &:active {
      background: ${theme.button.default.activeBg};
    }
  `;

export const subtleStyle = ({
  subtle,
  selected,
  theme,
  nature,
}: StyledButtonProps) =>
  subtle &&
  css`
    background: ${selected ? theme.button.subtle.bg : 'transparent'};
    color: ${nature === 'default'
      ? theme.button.subtle.color
      : natureColor(theme)[nature]};
    border: 1px solid transparent;

    &:hover {
      background: ${theme.button.subtle.hoverBg};
    }
  `;

export const flatStyle = ({
  flat,
  nature,
  theme,
}: StyledButtonProps) =>
  flat &&
  css`
    background: ${nature === 'default'
      ? theme.button.flat.bg
      : natureColor(theme)[nature]};
    color: ${nature === 'default'
      ? natureColor(theme).default
      : theme.button.flat.color};
    border-color: ${nature === 'default'
      ? theme.button.flat.bg
      : natureColor(theme)[nature]};

    &:hover {
      background: ${nature === 'default'
        ? theme.button.flat.hoverBg
        : lighten(0.1, natureColor(theme)[nature])};
      border-color: ${nature === 'default'
        ? theme.borderColor
        : lighten(0.1, natureColor(theme)[nature])};
    }
    &:active {
      background: ${nature === 'default'
        ? theme.button.flat.activeBg
        : darken(0.12, natureColor(theme)[nature])};
      border-color: ${nature === 'default'
        ? theme.button.flat.activeBorderColor
        : darken(0.12, natureColor(theme)[nature])};
    }
  `;

export const lightStyle = ({
  light,
  theme,
  nature,
}: StyledButtonProps) =>
  light &&
  css`
    background: ${theme.button.light.bg};
    color: ${natureColor(theme)[nature]};

    &:hover {
      background: ${theme.button.light.hoverBg};
    }
    &:active {
      background: ${theme.button.light.activeBg};
    }
  `;

export const linkStyle = ({ link, theme }: StyledButtonProps) =>
  link &&
  css`
    background: ${theme.button.link.bg};
    color: ${theme.button.link.color};
    border: 1px solid transparent;

    &:hover {
      background: ${theme.button.link.hoverBg};
      color: ${theme.button.link.hoverColor};
    }
  `;

export const linkDarkStyle = ({
  link,
  dark,
  theme,
}: StyledButtonProps) =>
  link &&
  dark &&
  css`
    color: ${theme.button.link.darkColor};

    &:hover {
      color: ${theme.button.link.hoverDarkColor};
    }
  `;

const underlineStyle = ({ underline }: StyledButtonProps) =>
  underline &&
  css`
    &:hover {
      text-decoration: underline;
    }
  `;

export const circleStyle = ({ circle }: StyledButtonProps) =>
  circle &&
  css`
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 0;
    border-radius: 100%;
    width: 2.066rem;
    line-height: 2.066rem;
    height: 2.066rem;
  `;

export const Button = styled('button').attrs(props => ({
  type: 'button',
  className: cn('vital__button', props.className),
}))<ButtonProps>`
  /* Basic */
  ${
    // @ts-ignore
    buttonBaseStyle
  }
  /* subtle */
  ${subtleStyle}
  /* flat */
  ${flatStyle}
  /* light */
  ${lightStyle}
  /* link */
  ${linkStyle}
  ${linkDarkStyle}

  /* underline */
  ${underlineStyle}

  /* circle */
  ${circleStyle}

  ${superBoxStyle}

  &[disabled] {
    color: ${({ theme }) => theme.button.disabled.color};
    border-color: ${({ theme }) => theme.button.disabled.borderColor};
    background: ${({ theme }) => theme.button.disabled.bg};
    cursor: not-allowed;
    pointer-events: none;
  }
  ${props => props.theme.button.buttonStyle};
`;

Button.defaultProps = {
  theme: defaultTheme,
  children: '',
  nature: 'default' as Nature,
  size: 'medium' as ButtonSize,
  circle: false,
  light: false,
  flat: false,
  link: false,
  subtle: false,
  selected: false,
  disabled: false,
};

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
    BoxProps {
  builtinSize?: BuiltinSize;
  /** 6 nature state */
  nature?: Nature;
  /** Flat style */
  flat?: boolean;
  /** Subtle style */
  subtle?: boolean;
  /** Light style */
  light?: boolean;
  /** Link style */
  link?: boolean;
  /** Button size */
  size?: ButtonSize;
  /** Circle button */
  circle?: boolean;
  /** Selected state */
  selected?: boolean;
  dark?: boolean;
  underline?: boolean;
  disabled?: boolean;
}

/**
 * @render react
 * @name Button
 * @description Button component with group
 * @example
 * <ButtonGroup>
 *   <Button nature="default" size="xlarge">Default</Button>
 *   <Button nature="primary" size="large">Default</Button>
 *   <Button nature="success" size="medium">Default</Button>
 *   <Button nature="info" size="small">Default</Button>
 *   <Button nature="alarm" size="xsmall">Default</Button>
 *   <Button nature="warning">Default</Button>
 *   <Button flat>Default</Button>
 *   <Button subtle>Default</Button>
 *   <Button light>Default</Button>
 *   <Button link>Default</Button>
 * </ButtonGroup>
 */
