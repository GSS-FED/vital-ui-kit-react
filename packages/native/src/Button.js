// @flow
/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import styled from 'styled-components/native';

const Root = styled.TouchableOpacity`
  padding-left: 13;
  padding-right: 13;
  height: ${props => props.height || 'auto'};
  justify-content: center;
  align-items: center;
  background-color: ${props =>
    props.active ? props.theme.tintColor : '#fff'};
  border-radius: 50;
  border-width: ${props => (props.active ? 0 : 0.5)};
  border-color: #fff;
`;

const Text = styled.Text`
  color: ${props => (props.active ? '#fff' : props.theme.tintColor)};
  font-size: 16;
`;

export type ButtonProps = {
  title: string,
  active?: boolean,
  onPress: () => void,
};

const Button = ({
  title,
  active,
  onPress,
  ...props
}: ButtonProps) => (
  <Root active={active} onPress={onPress} {...props}>
    <Text active={active}>{title}</Text>
  </Root>
);

export default Button;
