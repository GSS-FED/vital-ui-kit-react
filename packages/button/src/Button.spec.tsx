import React from 'react';
import { Button } from './Button';
import { render } from 'react-testing-library';

describe('Button', () => {
  it('should render with default styles', () => {
    const button = render(<Button>button text</Button>);
    expect(button).toMatchSnapshot();
  });
});
