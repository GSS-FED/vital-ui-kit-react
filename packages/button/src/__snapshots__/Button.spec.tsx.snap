// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render with default styles 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
  background: #F7F9FD;
  color: #456296;
  cursor: pointer;
  border-width: 1px;
  border-style: solid;
  border-color: #D8E2F5;
  font-size: 1rem;
  border-radius: 4px;
  padding: calc(0.533rem - 1px) 1.066rem;
  line-height: 1;
  font-weight: normal;
  margin: 0;
  outline: none;
  box-sizing: border-box;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  background: #D8E2F5;
}

.c0:active {
  background: #C3D1EB;
}

.c0[disabled] {
  color: #93a8ce;
  border-color: #eaeff8;
  background: #F7F9FD;
  cursor: not-allowed;
  pointer-events: none;
}

<body>
    <div>
      <button
        class="vital__button c0"
        type="button"
      >
        button text
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-htpNat vital__button japkjZ"
      type="button"
    >
      button text
    </button>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
